src/MizarBase/include/MizarBase/Address.h

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MIZAR_BASE_ADDRESS_H_
+#define MIZAR_BASE_ADDRESS_H_
+
+#include <functional>
+
+#include "OrbitBase/Typedef.h"
+
+namespace orbit_mizar_base {
+
+struct AddressTag {};
+using Address = orbit_base::Typedef<AddressTag, const uint64_t>;
```

Comment:
What's the purpose of making the inner type `const`? In other words, why does the type not be a regular value (which includes the ability to overwrite it)?

Reply:
> What's the purpose...

So we can't do `(*address)++`. We actually do the same for id-types. 

> why does the type not be a regular value (which includes the ability to overwrite it)

`Address` still supports assignment operator, the const over inner type merely means we turn-off the non-const getters.